cognite/client/_api/extractionpipelineruns.py

```diff
@@ -0,0 +1,84 @@
+from typing import *
+
+from cognite.client import utils
+from cognite.client._api_client import APIClient
+from cognite.client.data_classes import ExtractionPipelineRunFilter, ExtractionPipelineRunList
+from cognite.client.data_classes.extractionpipelineruns import ExtractionPipelineRun, StringFilter
+from cognite.client.data_classes.shared import TimestampRange
+
+
+class ExtractionPipelineRunsAPI(APIClient):
+    _RESOURCE_PATH = "/extpipes/runs"
+    _LIST_CLASS = ExtractionPipelineRunList
+
+    def list(
+        self,
+        external_id: str,
+        statuses: List[str] = None,
+        message_substring: str = None,
+        created_time: Union[Dict[str, Any], TimestampRange] = None,
+        limit: int = 25,
+    ) -> ExtractionPipelineRunList:
+        """List runs for an extraction pipeline with given external_id <>`_
+
+        Args:
+            external_id (str): Extraction pipeline external Id.
+            statuses (List[str]): success/failure/seen.
+            message_substring (str): failure message part.
+            created_time (int): The number of milliseconds since 00:00:00 Thursday, 1 January 1970, Coordinated Universal Time (UTC), minus leap seconds.
+            limit (int, optional): Maximum number of ExtractionPipelines to return. Defaults to 25. Set to -1, float("inf") or None
+                to return all items.
+
+        Returns:
+            ExtractionPipelineRunList: List of requested extraction pipeline runs
+
+        Examples:
+
+            List extraction pipeline runs::
+
+                >>> from cognite.client import CogniteClient
+                >>> c = CogniteClient()
+                >>> runsList = c.extraction_pipeline_runs.list(external_id="test ext id", limit=5)
+
+            Filter extraction pipeline runs on a given status::
+
+                >>> from cognite.client import CogniteClient
+                >>> c = CogniteClient()
+                >>> runsList = c.extraction_pipeline_runs.list(external_id="test ext id", statuses=["seen"], statuslimit=5)
+        """
+
```

Comment:
maybe add a sanitization check here on statuses (to confirm it is the among the three valid values?)

Reply:
Good idea!

Reply:
Or - is it necessary? The backend will return a 400 with an error anyway. This will just be yet another thing to change if we decide to add a fourth type of staus.

Reply:
no strong opinions on either, would save an api call though ;)